### PR TITLE
Add packs using packid

### PR DESCRIPTION
**Pull request by chaws**
Fix: https://github.com/Open-CMSIS-Pack/cpackget/issues/19

This PR does the following
1. Allows packID to be used to install packs:
1.1. Now the following pack add options are supported: `cpackget pack add ARM.mbedClient` or `cpackget pack add ARM.mbedClient.1.1.0` or `cpackget pack add http://www.keil.com/pack/ARM.mbedClient.1.1.0.pack` or `cpackget pack add ARM.mbedClient.1.1.0.pack`
2. It can find pack urls either using the pdsc URL + packID or via the url attribute in the release tag
2.1. If the url in the release tag end with some other file name, cpackget will make sure to name it to its proper pack file name when placing it under ".Download"
3. If a pack content is contained within a subfolder in the compressed file, cpackget will decompress that in the pack root folder, without the subfolder
3.1. If the pack content is under "compressedPack/contents", it'll be decompressed as "$CMSIS_PACK_ROOT/Vendor/PackName/x.y.z/contents", removing the "compressedPack" subfolder.

### CODE REVIEW

**Comment by jkrech** [comment]
Great stuff. Please address the failing windows test.

**Comment by chaws** [comment]
Yeah, Windows didn't like one of the renaming operations. Investigating why

File: cmd/installer/pack.go [hunk 1]
```diff
@@ -112,11 +124,19 @@ func (p *PackType) validate() error {
 	for _, file := range p.zipReader.File {
 		if filepath.Base(file.Name) == pdscFileName {
 
+			// Check if pack was compressed in a subfolder
+			subfoldersCount := strings.Count(file.Name, "/") + strings.Count(file.Name, "\\")
+			if subfoldersCount > 1 {
+				return errs.ErrPdscFileTooDeepInPack
```
Comment (jkrech) [inline cmd/installer/pack.go:130]: PdscFileTooDeepInPack semantically means that you found a pdsc but it is not at the right location, but actually you did not find a pdsc in the allowed locations
- root directory
- single sub directory

File: cmd/installer/root_test.go [hunk 2]
```diff
@@ -610,6 +643,385 @@ func TestAddPack(t *testing.T) {
 		packPath := packWithSubFolder
 		addPack(t, packPath, ConfigType{})
 	})
+
```
Comment (jkrech) [inline cmd/installer/root_test.go:646]: Should this be broken up further? Seems hard to read.

File: cmd/xml/pdsc.go [hunk 3]
```diff
@@ -52,6 +53,24 @@ func (p *PdscXML) LatestVersion() string {
 	return ""
 }
 
+// FindReleaseTagByVersion iterates over the PDSC file's releases tag and returns
+// the release that matching version.
+func (p *PdscXML) FindReleaseTagByVersion(version string) *ReleaseTag {
+	releases := p.ReleasesTag.Releases
```
Comment (jkrech) [inline cmd/xml/pdsc.go:59]: Is releases capable of comparing semantic versions?

**Review by jkrech — APPROVED** [review]

**Pull request merged**